serde-generate/runtime/typescript/lcs/lcsDeserializer.ts

```diff
@@ -0,0 +1,31 @@
+import leb from 'leb';
+import { BinaryDeserializer } from '../serde/binaryDeserializer';
+
+export class LcsDeserializer extends BinaryDeserializer {
+  constructor(data: Buffer) {
+    super(data);
+  }
+
+  public deserializeUleb128AsU32(): number {
```

Comment:
This one does not look right.

Reply:
could you elaborate?

Reply:
This code doesn't strip the highest bits added by the uleb encoding.

Reply:
the code by itself does not do the actual work. it's up to the "leb" library to do that. 
the thing is that "leb" must get a "clean" buffer with only the needed bytes for decoding. this loop is just for figuring out which bytes to put into that buffer and then let "leb" do the decode.

Reply:
Oh I see, my bad